mcs/class/test-helpers/SocketResponder.cs

```diff
@@ -91,8 +91,11 @@ private void Listen ()
 					listenSocket = tcpListener.AcceptSocket ();
 					listenSocket.Send (requestHandler (listenSocket));
 					try {
-						listenSocket.Shutdown (SocketShutdown.Receive);
+						// On Windows a Receive() is needed here before Shutdown() to consume the data some tests send.
+						listenSocket.ReceiveTimeout = 1000;
```

Comment:
Maybe it would be better to just do a `try-catch` around `Receive()`? If this hasn't been a problem on non-Windows before it shouldn't matter if the `Receive()` times out on those devices, right?